### ISSUE TITLE
[C4GT Community]: Set up comprehensive local development environment for AMRIT with updated developer documentation

### ISSUE THREAD

**Issue by drtechie** [PSMRI/AMRIT#18]
## Description

Digital Bharat Collaborative (DBC) is an initiative of Piramal Swasthya, the healthcare arm of the Piramal Foundation. DBC's focus areas include building a robust digital public health delivery platform, improving data-driven governance and empowering healthcare workers. AMRIT is an open-source platform designed to strengthen primary healthcare by empowering frontline health workers with the right digital tools for data-driven decision-making and seamless access to patients' Electronic Health Records (EHR) through ABDM-compliant, interoperable systems. 

To streamline the development process for AMRIT, we need a contributor to work on setting up a comprehensive local development environment. This setup should include all dependencies, such as Redis, MySQL databases, and sample data, along with configuration for running all AMRIT Spring Boot APIs and Angular frontend repositories locally. Proper developer documentation must also be updated to assist future contributors in replicating this setup efficiently. This task will enhance developer productivity and make onboarding new contributors smoother.

### What You Will Learn:

As a contributor, you will:
- Learn best practices for configuring local development environments.
- Gain experience in setting up and configuring backend services and frontend applications.
- Develop skills in writing developer-centric documentation.
- Collaborate with the AMRIT team to understand the structure and dependencies of the projects.

###  Desired Skills:

- Proficiency in configuring databases (e.g., PostgreSQL, MySQL) and related schema management.
- Experience with containerization tools (e.g., Docker) and dependency management.
- Familiarity with backend dependencies like Redis, Postgres, MySQL etc..
- Understanding of configuring Angular applications.
- Ability to document technical procedures clearly and effectively.

## Goals
- [ ] Set up a fully functional local environment for AMRIT with all dependencies, including Redis, databases, schema, and dummy data.
- [ ] Ensure that all AMRIT APIs are configured and bound to specific port numbers for local use.
- [ ] [Midpoint Milestone]
- [ ] Configure all Angular frontend repositories to run locally without errors.
- [ ] Update the developer documentation to include detailed steps for setting up the local environment, including configuration files, ports, and sample data.

### Expected Outcome

A complete local environment setup for AMRIT that includes all backend and frontend components. The updated developer documentation will help new contributors and developers replicate the setup with ease and minimize onboarding time.

### Acceptance Criteria

- All AMRIT APIs and Angular frontends run smoothly in a local development environment.
- Dependencies (Redis, database, MySQL DB schema, dummy data) are properly integrated and functioning.
- Clear, step-by-step developer documentation is provided and tested for accuracy.
- Developers can follow the documentation to set up the environment with minimal assistance.

### Implementation Details

The contributor will work with Technical Architect, Product and IT teams to understand the architecture and setup relevant local environment and documentation

### Mockups/Wireframes

NA

### Product Name

AMRIT

### Organisation Name

Piramal Swasthya Management Research Institute

### Domain

Healthcare

### Tech Skills Needed

Angular, Database, Debugging, DevOps, Docker, Spring Boot

### Mentor(s)

@drtechie

### Complexity

Medium

### Category

Configuration, Database, Documentation, Maintenance

### Bounty

This is a bounty ticket with a remuneration of INR 10,000.

### How to apply?

Submit [application form](https://forms.gle/8xJtt1LMCfVuQdYk9).

**Comment by Asp-Codes** [comment]
hey @drtechie can we have a talk on the API documentation it seems like it is not updated in the website!

**Comment by Shelly011s** [comment]
Hello, @drtechie I would like to work on this issue, there are several sections under developer doc that can be improved. Looking forward to connect on this.

**Comment by Sneha6003** [comment]
Hello @Asp-Codes  , are you still working on this issue? Kindly update

**Comment by Wahid7852** [comment]
I can help in doing this, seems fairly easy

**Comment by Asp-Codes** [comment]
> Hello @Asp-Codes  , are you still working on this issue? Kindly update

Yes I am actively working on this! There are Some PRs being raised!

**Comment by Sneha6003** [comment]
@Asp-Codes okay 👍

**Comment by drtechie** [comment]
@all-contributors please add @Asp-irin for code, doc, blog, infra

**Comment by drtechie** [comment]
@all-contributors please add @Asp-irin for code, doc, blog, infra

**Comment by drtechie** [comment]
Marking this issue as completed.
